benchexec/pytest_analyze_run_result.py

```diff
@@ -0,0 +1,197 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import sys
+import pytest
+import types
+
+from benchexec.util import ProcessExitCode
+from benchexec.model import Run
+from benchexec.result import (
+    RESULT_FALSE_REACH,
+    RESULT_ERROR,
+    RESULT_UNKNOWN,
+    RESULT_TRUE_PROP,
+)
+from benchexec.tools.template import BaseTool
+
+sys.dont_write_bytecode = True  # prevent creation of .pyc files
+
+normal_result = ProcessExitCode(raw=0, value=0, signal=None)
+
+
+@pytest.fixture(scope="class")
+def disable_non_critical_logging():
+    logging.disable(logging.CRITICAL)
```

Comment:
There was some discussion around this in #908, please have a look there. Ideal is a solution where
- we do not have to add something in every test file,
- log output does not clutter standard test execution (if everything works), and
- as much information as possible (including log messages) is shown for every failed test case.

IIRC we found some good way to do this in #908. If you see further possible improvements, please don't hesitate to add them.